aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1221,6 +1222,33 @@ static void addmm_impl_cpu_(
   }
 }
 
+// TODO: This should work through the structured kernel approach of TORCH_IMPL_FUNC
+// TODO: ensure devices are handled
+Tensor addmm_zerotensor(const Tensor& self, const Tensor& mat1, const Tensor& mat2, const Scalar& beta, const Scalar& alpha){
+  TORCH_INTERNAL_ASSERT(self._is_zerotensor() || mat1._is_zerotensor() || mat2._is_zerotensor() || beta.equal(0) || alpha.equal(0));
+  const auto self_sizes = self.sizes();
+  Tensor result;
+  // beta*self + alpha*(mat1@mat2) = 0
```

Comment:
we should perform the size and dim checks before this

Reply:
https://github.com/pytorch/pytorch/blob/5375b2e994ee666996a47526c7f9629e85acfc54/aten/src/ATen/native/LinearAlgebra.cpp#L1107-L1119

Reply:
I'd suggest creating a helper function that can be called from the cpu kernel as well as ZT kernel